### DISCUSSION TITLE
.NET MAUI on Android keeps not responding messages

### DISCUSSION THREAD

**Discussion by JawadJaber** [dotnet/maui#14820] [Q&A]
### Description

Hello
I have .NET MAUI application that do a lot of serialization and deserialization.
I have noticed this behavior when the app is in DEBUG mode where AOT is disabled and Enable trimming is disabled, that the app is keep sending messages either to wait and the app is not responding.

<img width="388" alt="image" src="https://user-images.githubusercontent.com/7624547/234916072-4ceaf9c7-c985-458e-8fb9-e977ba590c31.png">

<img width="278" alt="image" src="https://user-images.githubusercontent.com/7624547/234916241-799392ef-5c68-4db9-949f-a865f93283f9.png">

Any hint on how to handle this?

### Steps to Reproduce

.

### Link to public reproduction project repository

.

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android

### Affected platform versions

Android 11 and above

### Did you find any workaround?

.

### Relevant log output

```shell
.
```

**Comment by Xyncgas** [comment]
Can you please describe what kinds of messages it's sending
Because if I understood properly your question is running in Debug without AOT or Trimming you observe (how?) the messages being sent and making your app unresponsive

**Reply by JawadJaber** [reply]
I will send demonstration video shortly
Thanks for your message

**Reply by JawadJaber** [reply]
Sorry for the delay
Here is a short video shows the behavior
https://1drv.ms/v/s!AsZPe-3ZmkfLloZniQHrWWeKVswzYw?e=anT6qI

**Reply by JawadJaber** [reply]
@Redth
Hello man,
I have noticed that the cause of this issue is using AndHUD on android devices with progress. After I have used the progress HUD in the background it seems the library exhausts the mobile device, then after that the problem showed in the video starts happenning.. might be a memory leak or so.
But if I show the HUD without progress the problem is gone.

**Comment by EdCharbeneau** [comment]
I'm experimenting with running .NET MAUI on the Meta Quest 3 and running into the same issue. If I try to debug anything I get the "not responding" message whenever I hit a breakpoint. If you choose to "wait" you can debug successfully. It's really annoying though.

**Reply by Xyncgas** [reply]
I wonder how these platform determines the application is not responsive

**Comment by lobbo232** [comment]
This is happening regularly for me in android emulator, and it seems to happen on a release build too because Google keep rejecting my app for App loads, but not responsive with a screenshot of the wait/close message.

I know it does it on emulator, if you click "Wait" the app is still responding fine. If you run it on a device, the app seems to work fine also.

**Comment by FabriBertani** [comment]
This is happening to us as well, android Logcat is not throwing any relevant log either.

Any insights @samhouts @PureWeen @davidortinau @rachelkang @jsuarezruiz ?

**Comment by Pastajello** [comment]
This can happen when you have some hanged tasks running and never finishing, unfinished/hanged TaskCompletionSource.

**Reply by JawadJaber** [reply]
Can you describe more how to handle unfinished tasks?

**Comment by mackayn** [comment]
See this constantly in sim or hardware debugging on API35 constantly with the very latest VS2022 and android SDK's. Productivity killer and completely frustrating.

**Comment by FabriBertani** [comment]
@mackayn the workaround we found was to enabling debug app from `Settings -> Developer Options -> Select debug app` on both physical and emulator android. Although the problem did not recur (mostly because selecting debug app, the OS dedicate extra memory to it), I'm still not satisfied because we haven't been able to locate and fix the root of the problem yet.

**Reply by mackayn** [reply]
That does seem to work, big thanks as it's been driving me insane
